src/odemis/gui/comp/text.py

```diff
@@ -194,22 +194,32 @@ def onEnteredText(self, event):
                 self._showDropDown(False)
             event.Skip()
             return
-        found = False
 
+        found = None
         choices = self._choices
 
         for numCh, choice in enumerate(choices):
-            if self._match_function and self._match_function(text, choice):
-                found = True
+            if self._match_function:
+                if self._match_function(text, choice):
+                    found = numCh
+                    break
             elif choice.lower().startswith(text.lower()):
-                found = True
-            if found:
-                self._showDropDown(True)
-                item = self.dropdownlistbox.GetItem(numCh)
-                toSel = item.GetId()
-                self.dropdownlistbox.Select(toSel)
+                found = numCh
                 break
-        if not found:
+
+        # Try to match any part of the text if standard matching didn't return anything
+        if found is None:
+            for numCh, choice in enumerate(choices):
+                if text.lower() in choice.lower():
```

Comment:
You could move this if statement as a third `elif` statement in the for loop. I think then it would be more organized.

Reply:
As an `elif`, it'd have a different behaviour: it would find any text that matches anywhere, all the time, but we want it to happen only as a fallback. I've changed a little the code to make it clearer by use a for/else.